Sources/Views/MessagesCollectionView.swift

```diff
@@ -64,8 +64,19 @@ open class MessagesCollectionView: UICollectionView {
     // MARK: - Methods
 
     public func scrollToBottom(animated: Bool = false) {
-        guard let indexPath = indexPathForLastItem else { return }
-        scrollToItem(at: indexPath, at: .bottom, animated: animated)
+//        guard let indexPath = indexPathForLastItem else { return }
+//        scrollToItem(at: indexPath, at: .bottom, animated: animated)
+
+        // Work in progress here :)
+
+        guard indexPathForLastItem != nil else { return }
+
+        // This works fairly reliably, but using `DispatchQueue.main.async` might not let you invoke this method from an animation block :(
```

Comment:
@antonselyanin Just touching on somethings
- What were your concerns here, you mention "fairly reliably"?
- Why have we wrapped this in `DispatchQueue.main.async`?

Reply:
> What were your concerns here, you mention "fairly reliably"?
I don't remember exactly wrote it :(. Maybe there were case where it wouldn't scroll.

> Why have we wrapped this in DispatchQueue.main.async
It looks like we have to skip one run loop pass before we can invoke `layoutIfNeeded()`. UIKit needs this for some reason. At least, this is what I got from my experiments with this issue.

I think this is a dirty hack, which relies on unknown mysteries of UIKit (or, I just missing something).

IMO, this issue worth looking at: https://github.com/jessesquires/JSQMessagesViewController/issues/256

Reply:
Thanks for investigating this @antonselyanin. What do you think of [this approach?](https://github.com/MessageKit/MessageKit/issues/268#issuecomment-339680511)

I don't know if one approach has greater benefit over the other, but I'd like to clean up this PR and get this merged in.

Reply:
@SD10, I think the code from that comment might be an adaptation of this code: https://github.com/jessesquires/JSQMessagesViewController/blob/develop/JSQMessagesViewController/Controllers/JSQMessagesViewController.m#L386

Method `- (void)scrollToIndexPath:(NSIndexPath *)indexPath animated:(BOOL)animated`, line 386.

If this can be fixed without using `DispatchQueue.main.async` then the approach you linked to is better.

Reply:
Your solution seems to be more consistent but it's still not working all of the time 😓